meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -79,42 +79,31 @@ export const ShowStyleVariantsSettings = withTranslation()(
 					return
 				}
 
-				_.forEach(newVariants, (entry) => {
-					MeteorCall.showstyles
-						.insertShowStyleVariantWithBlueprint(
-							entry.showStyleBaseId,
-							entry.blueprintConfig,
-							entry._id,
-							entry._rundownVersionHash,
-							entry.name
-						)
-						.catch((error) => {
-							NotificationCenter.push(
-								new Notification(
-									undefined,
-									NoticeLevel.WARNING,
-									t('Failed to import Variant ' + entry.name + '. Make sure it is not already imported.', {
-										errorMessage: error + '',
-									}),
-									'VariantSettings'
-								)
-							)
-						})
-				})
+				this.importVariantsFromArray(newVariants)
 			}
 			reader.readAsText(file)
 		}
+
+		importVariantsFromArray = (variants: ShowStyleVariant[]) => {
+			const { t } = this.props
+			_.forEach(variants, (entry: ShowStyleVariant) => {
```

Comment:
Instead using a third party library to do a foreach loop, just use the foreach loop method on the Array:
`vairants.foreach((showStyleVariant: ShowStyleVariant) => {...})`

Sidenote, since your "entries" are ShowStyleVariants it would be nice if you named your `entry` for what it actually is i.e. `showStyleVariant`. Then you don't have to mental map that here entry means ShowStyleVariant, but in another loop in another file entry means BlueprintPiece.

Reply:
This has been implemented.